tests/conftest.py

```diff
@@ -120,6 +123,54 @@ def boolify(v):
     return result
 
 
+@pytest.fixture(scope="function")
+def api_default_testing_config(testing_config, request):
+    """Monkeypatch conda_build.api functions to use config=testing_config default
+
+    This requests fixture testing_config, thus implicitly testing_workdir, too.
+    """
+
+    # Allow single tests to disable this fixture even if outer scope adds it.
+    if "no_api_default_testing_config" in request.keywords:
+        return
+
+    arg_name = "config"
+    new_default = testing_config
+
+    saved_defaults = {}
+    saved_kwdefaults = {}
+
+    module = conda_build.api
+    for _, member in inspect.getmembers(module, inspect.isfunction):
+        if inspect.getmodule(member) is not module:
+            continue
+        arg_spec = inspect.getfullargspec(member)
+        defaults = member.__defaults__
+        kwdefaults = member.__kwdefaults__
+        try:
+            index = arg_spec.args[-len(defaults):].index(arg_name)
+        except (TypeError, ValueError):
+            pass
+        else:
+            saved_defaults[member] = defaults
+            member.__defaults__ = defaults[:index] + (new_default,) + defaults[index + 1:]
+        try:
+            kwdefaults[arg_name]
+        except (TypeError, KeyError):
+            pass
+        else:
+            saved_kwdefaults[member] = kwdefaults
+            member.__kwdefaults__ = {**kwdefaults, arg_name: new_default}
+
+    def restore_defaults():
+        for func, defaults in saved_defaults.items():
+            func.__defaults__ = defaults
+        for func, kwdefaults in saved_kwdefaults.items():
+            func.__kwdefaults__ = kwdefaults
+
+    request.addfinalizer(restore_defaults)
```

Comment:
This is way too magic and would be more obviously implemented by using pytest's `monkeypatch` fixture to overwrite `conda_buid.config.get_or_merge_config` to apply the test config.